core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -847,6 +847,7 @@ class DynamicListenerConfig(server: KafkaServer) extends BrokerReconfigurable wi
       newConfig.originals.asScala
         .filterKeys(_.startsWith(prefix))
         .filterKeys(k => !DynamicSecurityConfigs.contains(k))
+        .toMap
```

Comment:
How about `newConfig.originals.asScala.view.filterKeys(...).filterKeys(...).toMap`?

Reply:
Let's discuss this in the other comment about views. :)